thunder/dynamo/report.py

```diff
@@ -464,3 +471,194 @@ def helper_backend(gm, example_inputs):
+        folder = Path(folder)
+        folder.mkdir(exist_ok=True, parents=True)
+        nvfuser_callable = self.nvfusion_bsym._call_ctx[self.nvfusion_bsym.sym.name]
+        fd = nvfuser_callable.last_used
+
+        # The API for nvFuser version >=2.14
+        get_repro = getattr(fd, "repro_script_for", None)
+        # The legacy nvFuser API
+        if get_repro is None:
+            get_repro = getattr(fd, "getReproString", None)
+        if get_repro is None:
+            raise RuntimeError("The installed version of nvFuser does not support repro generation unless on crash.")
+
+        inputs = self.get_inputs()
+        repro_code = get_repro(inputs)
+        comment_str = f'"""\n{self.nvfusion_bsym}\n"""'
+
+        if file_name == None:
+            file_name = f"{self.name}_repro.py"
+        with open(folder / file_name, "w") as f:
+            print(comment_str, file=f)
+            print(repro_code, file=f)
+
+    def get_inputs(self):
+        return self.nvfusion_bsym._call_ctx[self.nvfusion_bsym.sym.name].last_inputs
+
+    def run_inductor_repro(self):
+        from thunder.executors.torch_compile import make_compiled
+
+        callable_for_torchcompile = make_compiled(
+            self.nvfusion_bsym.subsymbols, self.nvfusion_bsym.flat_args, self.nvfusion_bsym.flat_outs
+        )
+        inputs = self.get_inputs()
+        callable_for_torchcompile(*inputs)
+
+    def run_benchmark(self):
+        from thunder.dev_utils.utils import _benchmark_fusion_region_with_nvfuser_and_torch_compile
+
+        benchmark_comparison_data = _benchmark_fusion_region_with_nvfuser_and_torch_compile(self.nvfusion_bsym)
+        print(self.nvfusion_bsym)
+        print(benchmark_comparison_data)
+
+
+class ThunderFXGraphReport(FXGraphReport):
+    def __init__(self, gm, gm_name, subgraph_info, thunder_options):
+        super().__init__(gm, gm_name)
+        self.subgraph_info = subgraph_info
+        self.thunder_options = thunder_options
+
+        self.subgraph_reports: list[ThunderSegmentGraphReport] = []
+        self._create_subgraph_reports()
+
+    def get_thunder_module_name(self):
+        thunder_module_names = []
+        for node in self.subgraph_info.split_graph_module.graph.nodes:
+            target = node.target
+            if isinstance(target, str) and target.startswith("thunder_"):
+                thunder_module_names.append(f"{self.graph_name}_{target}")
+        return thunder_module_names
+
+    def _create_subgraph_reports(self):
+        thunder_module_names = self.get_thunder_module_name()
+        original_modules_to_thunder_modules = (
+            [m, compiled_m]
+            for m, compiled_m in self.subgraph_info.submodule_to_compiled_functions.items()
+            if compiled_m.compiler == CompilerType.THUNDER
+        )
+        example_inputs = self.subgraph_info.thunder_compiled_fns_example_inputs
+        self.split_reason = get_split_reasons_string(self.subgraph_info)
```

Comment:
Does it have to be an attribute on `self`? I think a normal variable should be sufficient here.